chaoscenter/graphql/server/pkg/authorization/roles.go

```diff
@@ -8,154 +8,101 @@ import (
-	GetYAMLData:                  {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
-	PredefinedWorkflowOperations: {MemberRoleOwnerString, MemberRoleEditorString},
-	ListPredefinedWorkflows:      {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
-	GetPredefinedExperimentYaml:  {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
-	ListDataSource:               {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
-	ListDashboard:                {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
-	GetGitOpsDetails:             {MemberRoleOwnerString},
-	ListWorkflowManifests:        {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
-	GetExperimentDetails:         {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
-	GetWorkflowManifestByID:      {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
-	ListImageRegistry:            {MemberRoleOwnerString},
-	GetImageRegistry:             {MemberRoleOwnerString},
-	CreateEnvironment:            {MemberRoleOwnerString, MemberRoleEditorString},
-	UpdateEnvironment:            {MemberRoleOwnerString, MemberRoleEditorString},
-	DeleteEnvironment:            {MemberRoleOwnerString, MemberRoleEditorString},
-	GetEnvironment:               {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
-	ListEnvironments:             {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
-	AddProbe:                     {MemberRoleOwnerString, MemberRoleEditorString},
-	UpdateProbe:                  {MemberRoleOwnerString, MemberRoleEditorString},
-	GetProbe:                     {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
-	ListProbes:                   {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
-	DeleteProbe:                  {MemberRoleOwnerString, MemberRoleEditorString},
+	ListInfrastructures:   {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
+	GetInfrastructure:     {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
+	GetManifest:           {MemberRoleOwnerString, MemberRoleEditorString},
+	GetInfraDetails:       {MemberRoleOwnerString, MemberRoleEditorString},
+	ListCharts:            {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
+	ListEnvironment:       {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
+	SaveChaosHub:          {MemberRoleOwnerString, MemberRoleEditorString},
+	CreateImageRegistry:   {MemberRoleOwnerString},
+	UpdateImageRegistry:   {MemberRoleOwnerString},
+	DeleteImageRegistry:   {MemberRoleOwnerString},
+	GetGitOpsDetails:      {MemberRoleOwnerString},
+	ListImageRegistry:     {MemberRoleOwnerString},
+	GetImageRegistry:      {MemberRoleOwnerString},
+	CreateEnvironment:     {MemberRoleOwnerString, MemberRoleEditorString},
```

Comment:
same

Reply:
The  `CreateEnvironment` should not be changed to `CreateExperiment`. It is a part of both [environment.resolver.go](https://github.com/litmuschaos/litmus/blob/master/chaoscenter/graphql/server/graph/environment.resolvers.go) and [chaos_experiment.resolver.go](https://github.com/litmuschaos/litmus/blob/master/chaoscenter/graphql/server/graph/chaos_experiment.resolvers.go). I've intentionally kept it in the `Chaos-Experiment` section.